pkg/envoy/xdsutil.go

```diff
@@ -216,6 +217,26 @@ func GetUpstreamTLSContext(downstreamIdentity identity.ServiceIdentity, upstream
 	return tlsConfig
 }
 
+// GetHTTP2ProtocolOptions creates an Envoy http configuration that matches the downstream protocol
+func GetHTTP2ProtocolOptions() map[string]*any.Any {
```

Comment:
Would return an error and let the caller handle it instead of silently filling a nil value in a struct field.